core/src/main/scala/kafka/server/DelayedRemoteFetch.scala

```diff
@@ -36,6 +37,7 @@ import scala.collection._
  * in the remote fetch operation purgatory
  */
 class DelayedRemoteFetch(remoteFetchTask: Future[Void],
+                         remoteLogReader: RemoteLogReader,
```

Comment:
This is an existing issue. It would be better if we could consolidate remoteFetchTask and remoteFetchResult into a single future.

Reply:
Good idea! I'm giving a try to this by defining a `RemoteFetchFuture` that includes both futures and the RemoteLogReader to handle all the cancel procedure.